### ISSUE TITLE
Add social media share buttons

### ISSUE THREAD

**Issue by scytherswings** [PinsterTeam/Frontend#102]

**Comment by scytherswings** [comment]
We need to provide users an easy way to share pins to social media sites. There are several react packages out there that help provide this functionality but none really do it well, or I suck at implementing them.

I have a branch https://github.com/PinsterTeam/Frontend/tree/newell/social_media_share which can be used as a starting point, I think I got most of the button generation working I just didn't have the CSS figured out. 

Also we'll need to generate the URLs to link back, but this feature isn't really useful until a user can link back to our page and have a specific pin open up.. perhaps we can hack this to link to a search query for that pin, or just the show page for the pin itself.

This should be a pretty fast project for someone familiar with styling and stuff.

**Comment by markmoser** [comment]
I think we should nail down the pin-page/modal discussions before this gets too far in dev. There were discussions on making the modals use pin-page urls, etc but I don't think it ever made it to a ticket.

**Comment by stretch1414** [comment]
Agreed. I think we need to flesh out and clean up the modal a bit more and
then tackle this once that all makes more sense.
